src/tslint.ts

```diff
@@ -43,9 +44,20 @@ module Lint {
             this.fileName = fileName;
             this.source = source;
             this.options = options;
+
+            if (this.options.allowNonTsExtensions == null) {
+                this.options.allowNonTsExtensions = false;
+            }
         }
 
         public lint(): LintResult {
+            const tsExtensions = [".ts"];
+            const fileExtension = path.extname(this.fileName);
+            const isTsFile = (tsExtensions.indexOf(fileExtension) >= 0);
+            if (!(isTsFile || this.options.allowNonTsExtensions)) {
```

Comment:
personally I would find this more readable as `if (!isTsFile && !this.options.allowNonTsExtensions)`